src/Umbraco.Web/Editors/CodeFileController.cs

```diff
@@ -0,0 +1,155 @@
+                    {
+                        Services.FileService.DeleteScript(virtualPath, Security.CurrentUser.Id);
+                        return Request.CreateResponse(HttpStatusCode.OK);
+                    }
+                    else
+                    {
+                        return Request.CreateErrorResponse(HttpStatusCode.NotFound, "No Script found with the specified path");
+                    }
+                    break;
+                default:
+                    throw new ArgumentException("File Type not supported", "type");
+            }
+        }
+
+        public CodeFileDisplay PostSave(string type, CodeFileDisplay display)
+        {
+            if (display == null)
+            {
+                throw new ArgumentNullException("No file object has been passed");
+            }
+            else
+            {
+                switch (type)
+                {
+                    case "partialView":
+                        var view = Services.FileService.GetPartialView(display.VirtualPath);
+                        if (view != null)
+                        {
+                            view.Content = display.Content;
+                            Services.FileService.SavePartialView(view, Security.CurrentUser.Id);
+                        }
+                        else
+                        {
+                            throw new ArgumentNullException($"File doesn't exist - {display.VirtualPath}");
+                        }
+                        break;
+                    case "partialViewMacro":
+                        var viewMacro = Services.FileService.GetPartialViewMacro(display.VirtualPath);
+                        if (viewMacro != null)
+                        {
+                            viewMacro.Content = display.Content;
+                            Services.FileService.SavePartialViewMacro(viewMacro, Security.CurrentUser.Id);
+                        }
+                        else
+                        {
+                            throw new ArgumentNullException($"File doesn't exist - {display.VirtualPath}");
+                        }
+                        break;
+                    case "script":
+                        var script = Services.FileService.GetScriptByName(display.VirtualPath);
+                        if (script != null)
+                        {
+                            script.Content = display.Content;
+                            Services.FileService.SaveScript(script, Security.CurrentUser.Id);
+                        }
+                        else
+                        {
+                            throw new ArgumentNullException($"File doesn't exist - {display.VirtualPath}");
+                        }
+                        break;
+                    default:
+                        throw new ArgumentException("File Type not supported", "type");
+                }
+                return display;
```

Comment:
In all `PostSave` methods we also display a success notification, you can see this in all other controllers like: `display.AddSuccessNotification(ui.Text("speechBubbles", "dataTypeSaved"), "");` which means that your model also needs to be of type `INotificationModel`

Similarly, when POSTing, developers can cancel events and add their own messages as to why it's been canceled, these messages need to be bubbled up to the UI. To do this you should be inheriting from `BackOfficeNotificationsController`